stactools_sentinel2/stactools/sentinel2/utils.py

```diff
@@ -0,0 +1,83 @@
+import os
+from typing import Callable, Optional
+
+from lxml import etree
+from pystac import (STAC_IO, STACError)
+import orjson
+
+# Type alias
+ReadHrefModifier = Callable[[str], str]
+
+
+def read_xml(href: str, read_href_modifier: Optional[ReadHrefModifier] = None):
+    if read_href_modifier is None:
+        text = STAC_IO.read_text(href)
+    else:
+        text = STAC_IO.read_text(read_href_modifier(href))
+    return etree.fromstring(bytes(text, encoding='utf-8'))
+
+
+def convert(typ: Callable, v: Optional[str]) -> Optional[int]:
+    return v if v is None else typ(v)
+
+
+def band_index_to_name(index):
+    raw_band_id = index + 1
+    if raw_band_id == 9:
+        band_id = '8A'
+    elif raw_band_id > 9:
+        band_id = str(raw_band_id - 1)
+    else:
+        band_id = str(raw_band_id)
+    return f'B{band_id.rjust(2, "0")}'
+
+
+def extract_gsd(image_path: str) -> float:
+    return float(image_path[-7:-5])
+
+
+def get_xml_node(root, xpath):
+    return root.find(xpath, root.nsmap)
+
+
+def list_xml_node(root, xpath):
+    return root.findall(xpath, root.nsmap)
+
+
+def get_xml_node_text(root: etree.ElementTree, xpath: str) -> Optional[str]:
+    node = root.find(xpath, root.nsmap)
+    if node is not None:
+        return node.text
+    return None
+
+
+def get_xml_node_attr(root: etree.ElementTree,
+                      attr: str,
+                      xpath=None) -> Optional[str]:
+    node = root
+    if xpath is not None:
+        node = root.find(xpath, root.nsmap)
+
+    if node is not None:
+        return node.get(attr)
+
+    return None
+
+
+def orjson_save_json(uri, json_dict):
+    dirname = os.path.dirname(uri)
+    if dirname != '' and not os.path.isdir(dirname):
+        os.makedirs(dirname)
+    with open(uri, 'wb') as f:
+        f.write(orjson.dumps(json_dict, option=orjson.OPT_INDENT_2))
+
+
+def orjson_save_item(item, include_self_link=True, dest_href=None):
```

Comment:
Does this work with remote paths?

Using the pystac save method takes advantage of fsspec (set via [here](https://github.com/stac-utils/stactools/blob/master/stactools_core/stactools/core/io.py#L5)) for STAC_IO, which allows users to plug in different file systems and write directly to remote storage. While the speedup is nice, if this only writes to local we'd want to make the usage of orjson is wrapped in an option at least.

Reply:
Which reminds me that [this line](https://github.com/stac-utils/stactools/blob/master/stactools_naip/stactools/naip/__init__.py#L7) is missing from the subpackage; can you put that in and then make a note in the README section about adding new datasets that this is one of the steps? This is required to enable fsspec usage in the subpackage.